scripts/download_checkpoints.py

```diff
@@ -0,0 +1,137 @@
+import csv
+import os
+import requests
+from tqdm import tqdm
+import argparse
+from pathlib import Path
+from urllib.parse import urljoin
+
+def convert_to_r2_url(http_url):
+    """Convert HTTP URL to R2 URL format."""
+    if http_url.startswith('https://olmo-checkpoints.org/'):
+        return http_url.replace('https://olmo-checkpoints.org/', 'r2://olmo-checkpoints/')
+    return http_url
+
+def convert_to_public_url(r2_url):
+    """Convert R2 URL to public HTTP URL format."""
+    if r2_url.startswith('r2://olmo-checkpoints/'):
+        return r2_url.replace('r2://olmo-checkpoints/', 'https://olmo-checkpoints.org/')
+    return r2_url
+
+def download_file(url, save_path, chunk_size=8192):
+    """Download a file with progress bar."""
+    response = requests.get(url, stream=True)
+    response.raise_for_status()
+    total_size = int(response.headers.get('content-length', 0))
+    save_path.parent.mkdir(parents=True, exist_ok=True)
+    
+    with open(save_path, 'wb') as f:
+        with tqdm(total=total_size, unit='B', unit_scale=True, desc=save_path.name) as pbar:
+            for chunk in response.iter_content(chunk_size=chunk_size):
+                if chunk:
+                    f.write(chunk)
+                    pbar.update(len(chunk))
```

Comment:
Retries should be handled in here. If the request fails with a recoverable error code ([anything that starts with 5XX, 408, 409, 429](https://en.wikipedia.org/wiki/List_of_HTTP_status_codes)), it should wait one second, and then try again. Try up to 5 times, and then give up. When giving up, it must make sure that the file at `save_path` does not exist.